build/util/src/lib.rs

```diff
@@ -215,3 +235,52 @@ fn toml_from_env<T: DeserializeOwned>(var: &str) -> Result<Option<T>> {
         .context("deserializing configuration")?;
     Ok(Some(rval))
 }
+
+pub fn build_notifications() -> Result<()> {
+    let out_dir = out_dir();
+    let dest_path = out_dir.join("notifications.rs");
+    let mut out = std::fs::File::create(dest_path)?;
+
+    let full_task_config = task_full_config_toml()?;
+
+    if full_task_config.notifications.len() >= 32 {
+        bail!(
+            "Too many notifications; \
+             overlapping with `INTERNAL_TIMER_NOTIFICATION`"
+        );
+    }
+    if full_task_config.name == "task-jefe"
+        && full_task_config.notifications.get(0).cloned()
+            != Some("fault".to_string())
+    {
+        bail!("`jefe` must have \"fault\" as its first notification");
```

Comment:
Is this for compatibility with `humility jefe`?

Reply:
Nope, it's for compatibility with the kernel, which always posts faults to the supervisor's bit 0.